controllers/networkplugins_controller.go

```diff
@@ -17,35 +17,480 @@ limitations under the License.
 package controllers
```

Comment:
Note after reviewing: with the controller logic itself already likely becoming pretty big. You might want to consider moving all the template rendering stuff to a separate package under `pkg` 🙂